legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/classpath/ClasspathUsingMavenFactory.java

```diff
@@ -446,28 +429,45 @@ private URLClassLoader createClassloader(File maven, File pom, File settingXml,
             LOGGER.error("Unable to construct extensions classpath", e);
             return null;
         }
+        finally
+        {
+            try
+            {
+                if (tempPom != null)
+                {
+                    Files.delete(tempPom);
+                }
+            }
+            catch (IOException e)
+            {
+                LOGGER.error("Failed to delete temp pom", e);
+            }
+        }
     }
 
-    private File computeExtraDependenciesPom(File maven, File pom, File settingXml, List<String> extraDependencies, List<SDLCPlatform> platformVersions) throws IOException, MavenInvocationException
+    private Path addExtraDependenciesToPom(File maven, Path pom, File settingXml, List<String> extraDependencies, List<SDLCPlatform> platformVersions) throws IOException, MavenInvocationException
     {
-        File legendLspExtraElementsPom = File.createTempFile("pom", ".xml");
-        legendLspExtraElementsPom.deleteOnExit();
-
-        try (OutputStream pomOs = Files.newOutputStream(legendLspExtraElementsPom.toPath()))
+        Model model;
+        try (InputStream is = Files.newInputStream(pom))
         {
-            String engineVersion = this.findEngineVersion(maven, pom, settingXml, platformVersions);
+            model = new MavenXpp3Reader().read(is);
+        }
+        catch (Exception e)
+        {
+            throw new IOException("Cannot read pom model", e);
+        }
 
-            Model extraDependenciesModel = new Model();
-            extraDependenciesModel.setModelVersion("4.0.0");
-            extraDependenciesModel.setGroupId("org.finos.legend.ide.lsp");
-            extraDependenciesModel.setArtifactId("default-pom");
-            extraDependenciesModel.setVersion("0.0.1-SNAPSHOT");
+        Path tempPom = pom.getParent().resolve("legend_extension_temp_pom.xml");
```

Comment:
I might use Files.createTempFile here to avoid possible conflicts.

Reply:
Done - make sense in case to prevent conflicts in case delete fail on prev execution.